src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -549,90 +553,75 @@ export class EditorService extends Disposable implements EditorServiceImpl {
-			untypedEditor.options = editor.options ?? untypedEditor.options;
 		}
 
 		// Untyped: take as is
 		else {
 			untypedEditor = editor;
 		}
 
+		// Typed editors that cannot convert to untyped will be taken
+		// as is without override. We still want to resolve a group though.
+		if (!untypedEditor) {
+			const [group, activation] = this.findTargetGroup(editor, preferredGroup);
+
+			return [OverrideStatus.NONE, group, activation];
+		}
+
 		// We need a `override` for the untyped editor if it is
 		// not there so we call into the editor override service
-		let conflictingDefaults = false;
+		let hasConflictingDefaults = false;
 		if (typeof untypedEditor.options?.override !== 'string') {
 			const populatedInfo = await this.editorOverrideService.populateEditorId(untypedEditor);
 			if (!populatedInfo) {
 				return [OverrideStatus.ABORT, undefined, undefined]; // we could not resolve the editor id
 			}
 
-			conflictingDefaults = populatedInfo.conflictingDefault;
+			hasConflictingDefaults = populatedInfo.conflictingDefault;
 		}
 
 		// Find the target group for the editor
 		const [group, activation] = this.findTargetGroup(untypedEditor, preferredGroup);
 
-		// Adjust options based on group activation if any
-		if (activation) {
-			untypedEditor.options = { ...untypedEditor.options, activation };
-		}
-
-		return [await this.editorOverrideService.resolveEditorInput(untypedEditor, group, conflictingDefaults), group, untypedEditor.options];
+		return [await this.editorOverrideService.resolveEditorInput(untypedEditor, group, hasConflictingDefaults), group, activation];
 	}
 
-	private findTargetGroup(editor: IEditorInput, options?: IEditorOptions, preferredGroup?: OpenInEditorGroup): [IEditorGroup, EditorActivation | undefined];
-	private findTargetGroup(editor: IUntypedEditorInput, preferredGroup?: OpenInEditorGroup): [IEditorGroup, EditorActivation | undefined];
-	private findTargetGroup(editor: IEditorInput | IUntypedEditorInput, optionsOrPreferredGroup?: IEditorOptions | OpenInEditorGroup, preferredGroup?: OpenInEditorGroup): [IEditorGroup, EditorActivation | undefined];
-	private findTargetGroup(editor: IEditorInput | IUntypedEditorInput, optionsOrPreferredGroup?: IEditorOptions | OpenInEditorGroup, preferredGroup?: OpenInEditorGroup): [IEditorGroup, EditorActivation | undefined] {
-		let group: IEditorGroup | undefined;
-
-		// Typed Editor Support
-		let options: IEditorOptions | undefined;
-		if (isEditorInput(editor)) {
-			options = optionsOrPreferredGroup as IEditorOptions;
-
-			group = this.doFindTargetGroup(editor, options, preferredGroup);
-		}
-
-		// Untyped Editor Support
-		else {
-			options = editor.options;
-
-			preferredGroup = optionsOrPreferredGroup as OpenInEditorGroup;
-			group = this.doFindTargetGroup(editor, options, preferredGroup);
-		}
+	private findTargetGroup(editor: IEditorInputWithOptions, preferredGroup: OpenInEditorGroup | undefined): [IEditorGroup, EditorActivation | undefined];
```

Comment:
I changed this method to work with `IEditorInputWithOptions` to make the overloads less complicated.